src/request_weaver.cc

```diff
@@ -31,10 +35,131 @@ namespace transcoding {
 namespace pb = google::protobuf;
 namespace pbconv = google::protobuf::util::converter;
 
+namespace {
+
+pb::util::Status bindingFailureStatus(internal::string_view field_name,
+                                      internal::string_view type,
+                                      const pbconv::DataPiece& value) {
+  return pb::util::Status(
+      pb::util::StatusCode::kInvalidArgument,
+      pb::StrCat("Failed to convert binding value ", field_name, ":",
+                 value.ValueAsStringOrDefault(""),
+                 " to ", type));
+}
+
+pb::util::Status isEqual(internal::string_view field_name,
+                         const pbconv::DataPiece& value_in_body,
+                         const pbconv::DataPiece& value_in_binding) {
+  bool value_is_same = true;
+  switch (value_in_body.type()) {
+    case pbconv::DataPiece::TYPE_INT32: {
+      pb::util::StatusOr<int32_t> status = value_in_binding.ToInt32();
+      if (!status.ok()) {
+        return bindingFailureStatus(field_name, "int32", value_in_binding);
+      }
+      if (status.value() != value_in_body.ToInt32().value())
```

Comment:
I remember c++ style guild requires 

if (expr) {
   one_line_code;
}

Reply:
SG. Added parenthesis for this line and other lines as well.